components/Auth/index.tsx

```diff
@@ -1,45 +1,142 @@
+import { useState, useMemo, ChangeEvent, MouseEventHandler } from 'react'
+import { SubmitHandler } from 'react-hook-form'
 import { Button, Form, TextField, Modal } from '@components/Custom'
+import { selectSet, Character, GridField } from '@components/Grid/GridForm'
 
+const CHARACTERS_FOR_AUTH = 3
 interface AuthProps {
   open: boolean
   onClose: () => void
 }
 
 interface FormValues {
   username: string
-  DOB: string
+  password: Character[]
 }
 
 const defaultValues: FormValues = {
   username: '',
-  DOB: ''
+  password: []
 }
 
 const Auth = (props: AuthProps) => {
+  const [input, setInput] = useState('')
+  const [username, setUsername] = useState('')
+  const grid = useMemo<Character[]>(() => selectSet(username), [username])
+
+  const handleUsernameChange = (
+    e: ChangeEvent<HTMLInputElement>,
+    values: 'username'
+  ) => {
+    setInput(e.target.value)
+    // console.log(e.target.value)
+    // console.log(JSON.stringify(defaultValues))
+  }
+
+  const handleUsernameSubmit: MouseEventHandler<HTMLButtonElement> = () => {
+    setUsername(input)
+    // console.log(input)
+    // console.log(defaultValues)
+  }
+
+  const handlePasswordSubmit: SubmitHandler<FormValues> = value => {
+    if (value.password.length !== CHARACTERS_FOR_AUTH) {
+      return
+    }
+    const newPassword = value.password.join('')
+    console.log(username)
+    console.log(value.password)
+    console.log(newPassword)
```

Comment:
I'd recommend removing console logs before committing to avoid cluttering code and the console for other developers